test/configuration/displayname.spec.conf/displayname_schema.json

```diff
@@ -0,0 +1,780 @@
+            {
+              "table_name": "accommodation",
+              "schema_name": "displayname_schema",
+              "column_name": "thumbnail"
+            }
+          ],
+          "annotations": {
+            "comment": ["thumbnail"]
+          },
+          "referenced_columns": [
+            {
+              "table_name": "file",
+              "schema_name": "displayname_schema",
+              "column_name": "id"
+            }
+          ]
+        },
+        {
+          "comment": null,
+          "foreign_key_columns": [
+            {
+              "table_name": "accommodation",
+              "schema_name": "displayname_schema",
+              "column_name": "cover"
+            }
+          ],
+          "annotations": {
+            "comment": ["thumbnail"]
+          },
+          "referenced_columns": [
+            {
+              "table_name": "file",
+              "schema_name": "displayname_schema",
+              "column_name": "id"
+            }
+          ]
+        }
+      ],
+      "table_name": "accommodation",
+      "schema_name": "displayname_schema",
+      "column_definitions": [
+        {
+          "comment": null,
+          "name": "id",
+          "default": null,
+          "nullok": false,
+          "type": {
+            "typename": "int4"
+          },
+          "annotations": {
+            "comment": [
+              "hidden"
+            ]
+          }
+        },
+        {
+          "comment": null,
+          "name": "title",
+          "default": null,
+          "nullok": false,
+          "type": {
+            "typename": "text"
+          },
+          "annotations": {
+            "comment": [
+              "title",
+              "top",
+              "orderby"
+            ],
+            "description": {
+              "display": "Name of Accommodation"
+            },
+            "facetOrder": [
+              "1"
+            ]
+          }
+        },
+        {
+          "comment": null,
+          "name": "website",
+          "default": null,
+          "nullok": true,
+          "type": {
+            "typename": "text"
+          },
+          "annotations": {
+            "comment": [
+              "url"
+            ],
+            "description": {
+              "display": "Website"
+            }
+          }
+        },
+        {
+          "comment": null,
+          "name": "category",
+          "default": null,
+          "nullok": false,
+          "type": {
+            "typename": "text"
+          },
+          "annotations": {
+            "comment": ["top"],
+            "description": {
+              "display": "Category"
+            },
+            "facetOrder": [
+              "2"
+            ]
+          }
+        },
+        {
+          "comment": null,
+          "name": "rating",
+          "default": 3,
+          "nullok": false,
+          "type": {
+            "typename": "float4"
+          },
+          "annotations": {
```

Comment:
As a general practice, what I think we should all be doing is just creating disctint schemas specific to one test suite. Its okay if it covers a number of test cases but they should be a cohesive set of tests. And this is what I would want the tutorial to point out to other developers. Then for example, if I am adding a feature to displayname handling and need to extend the test case. I can “easily” look at the schema and understand how to make use of it for my test, and how to extend it if it didn’t have what I needed for my test.

From: Aref Shafaei <notifications@github.com>
Reply-To: informatics-isi-edu/ermrestjs <reply@reply.github.com>
Date: Wednesday, July 13, 2016 at 11:11 AM
To: informatics-isi-edu/ermrestjs <ermrestjs@noreply.github.com>
Cc: Robert Schuler <schuler@isi.edu>, Assign <assign@noreply.github.com>
Subject: Re: [informatics-isi-edu/ermrestjs] Displayname Unit Tests (#82)


In test/configuration/displayname.spec.conf/displayname_schema.json<https://github.com/informatics-isi-edu/ermrestjs/pull/82#discussion_r70678503>:

> +              "display": "Category"

> +            },

> +            "facetOrder": [

> +              "2"

> +            ]

> +          }

> +        },

> +        {

> +          "comment": null,

> +          "name": "rating",

> +          "default": 3,

> +          "nullok": false,

> +          "type": {

> +            "typename": "float4"

> +          },

> +          "annotations": {

I copy/pasted the schema that has been created before and just added the tag:misd.isi.edu,2015:display annotation for some specific tables. I'll get rid of this unused annotations.

—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub<https://github.com/informatics-isi-edu/ermrestjs/pull/82/files/c62508c3a60587317c1942206077d66e60dbe07f#r70678503>, or mute the thread<https://github.com/notifications/unsubscribe/AEAhWhJHfubHtT0ma2dOcVbfY6wiPbkbks5qVSo4gaJpZM4JK9IH>.